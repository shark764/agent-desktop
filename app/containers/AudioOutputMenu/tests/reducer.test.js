/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

import AudioOutputSlice from '../reducer';
const { actions, reducer } = AudioOutputSlice;

const mockState = {
  availableOutputDevices: [],
  activeOutputRingtoneDevice: 'default',
  activeOutputSpeakerDevice: 'default',
  activeOutputNotificationDevice: 'default',
  isOutputSelectionSupported: false,
};

const availableOutputDevices = [
  { deviceId: 'default', label: 'Default' },
  {
    deviceId:
      'aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b',
    label: 'Built-in Audio Analog Stereo',
  },
  {
    deviceId:
      '6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb',
    label: 'Plantronics C320 Analog Stereo',
  },
  {
    deviceId:
      '807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957',
    label: 'USG PnIXAu20o  Analog Stereo',
  },
];

describe('reducer', () => {
  it('sets the correct initial state', () => {
    expect(reducer(mockState, {})).toMatchSnapshot();
  });

  describe('audioOutputPreferences/setAvailableOutputDevices', () => {
    it('sets new available output devices', () => {
      expect(
        reducer(
          mockState,
          actions.setAvailableOutputDevices(availableOutputDevices)
        )
      ).toMatchSnapshot();
    });
  });

  describe('audioOutputPreferences/setActiveOutputRingtoneDevice', () => {
    it('sets new active ringtone devices', () => {
      expect(
        reducer(
          mockState,
          actions.setActiveOutputRingtoneDevice('mock ringtone id')
        )
      ).toMatchSnapshot();
    });
  });

  describe('audioOutputPreferences/setActiveOutputSpeakerDevice', () => {
    it('sets new active speaker devices', () => {
      expect(
        reducer(
          mockState,
          actions.setActiveOutputSpeakerDevice('mock speaker id')
        )
      ).toMatchSnapshot();
    });
  });

  describe('audioOutputPreferences/setActiveOutputNotificationDevice', () => {
    it('sets new active notification device', () => {
      expect(
        reducer(
          mockState,
          actions.setActiveOutputNotificationDevice('mock notification id')
        )
      ).toMatchSnapshot();
    });
  });

  describe('audioOutputPreferences/setOutputSelectionSupported', () => {
    it('sets bool value to support feature control variable ', () => {
      expect(
        reducer(mockState, actions.setOutputSelectionSupported(true))
      ).toMatchSnapshot();
    });
  });
});
