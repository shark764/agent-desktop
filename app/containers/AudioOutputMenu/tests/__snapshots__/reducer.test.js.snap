// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer audioOutputPreferences/setActiveOutputNotificationDevice sets new active notification device 1`] = `
Object {
  "activeOutputNotificationDevice": "mock notification id",
  "activeOutputRingtoneDevice": "default",
  "activeOutputSpeakerDevice": "default",
  "availableOutputDevices": Array [],
  "isOutputSelectionSupported": false,
}
`;

exports[`reducer audioOutputPreferences/setActiveOutputRingtoneDevice sets new active ringtone devices 1`] = `
Object {
  "activeOutputNotificationDevice": "default",
  "activeOutputRingtoneDevice": "mock ringtone id",
  "activeOutputSpeakerDevice": "default",
  "availableOutputDevices": Array [],
  "isOutputSelectionSupported": false,
}
`;

exports[`reducer audioOutputPreferences/setActiveOutputSpeakerDevice sets new active speaker devices 1`] = `
Object {
  "activeOutputNotificationDevice": "default",
  "activeOutputRingtoneDevice": "default",
  "activeOutputSpeakerDevice": "mock speaker id",
  "availableOutputDevices": Array [],
  "isOutputSelectionSupported": false,
}
`;

exports[`reducer audioOutputPreferences/setAvailableOutputDevices sets new available output devices 1`] = `
Object {
  "activeOutputNotificationDevice": "default",
  "activeOutputRingtoneDevice": "default",
  "activeOutputSpeakerDevice": "default",
  "availableOutputDevices": Array [
    Object {
      "deviceId": "default",
      "label": "Default",
    },
    Object {
      "deviceId": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
      "label": "Built-in Audio Analog Stereo",
    },
    Object {
      "deviceId": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
      "label": "Plantronics C320 Analog Stereo",
    },
    Object {
      "deviceId": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
      "label": "USG PnIXAu20o  Analog Stereo",
    },
  ],
  "isOutputSelectionSupported": false,
}
`;

exports[`reducer audioOutputPreferences/setOutputSelectionSupported sets bool value to support feature control variable  1`] = `
Object {
  "activeOutputNotificationDevice": "default",
  "activeOutputRingtoneDevice": "default",
  "activeOutputSpeakerDevice": "default",
  "availableOutputDevices": Array [],
  "isOutputSelectionSupported": true,
}
`;

exports[`reducer sets the correct initial state 1`] = `
Object {
  "activeOutputNotificationDevice": "default",
  "activeOutputRingtoneDevice": "default",
  "activeOutputSpeakerDevice": "default",
  "availableOutputDevices": Array [],
  "isOutputSelectionSupported": false,
}
`;
